src/key.cpp

```diff
@@ -263,6 +263,7 @@ bool CKey::SignCompact(const uint256 &hash, std::vector<unsigned char>& vchSig)
 
 bool CKey::SignSchnorr(const uint256& hash, Span<unsigned char> sig, const uint256* merkle_root, const uint256* aux) const
```

Comment:
Instead of changing the meaning of `aux==nullptr` here, what about:

- Remove the default arguments from `CKey::SignSchnorr` (they're unused)
- Disallow nullptr by changing the argument to a `const uint256& aux`
- Pass zero in from the caller if it wants to use zero

This seems more explicit and clear to me? (The behavior is not documented in the doc comment, in any case, right now)

Reply:
Done.